library/md5.c

```diff
@@ -314,7 +314,11 @@ int mbedtls_md5_finish(mbedtls_md5_context *ctx,
     MBEDTLS_PUT_UINT32_LE(ctx->state[2], output,  8);
     MBEDTLS_PUT_UINT32_LE(ctx->state[3], output, 12);
 
-    return 0;
+    ret = 0;
+
+exit:
+    mbedtls_md5_free(ctx);
```

Comment:
(aside) I really wish our functions that clear but don't `free()` memory weren't named `mbedtls_foo_free()`

Reply:
https://github.com/Mbed-TLS/mbedtls/issues/7838